include/multipass/availability_zone_manager.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_AVAILABILITY_ZONE_MANAGER_H
+#define MULTIPASS_AVAILABILITY_ZONE_MANAGER_H
+
+#include "availability_zone.h"
+
+namespace multipass
+{
+class AvailabilityZoneManager : DisabledCopyMove
+{
+public:
+    using UPtr = std::unique_ptr<AvailabilityZoneManager>;
+    using ShPtr = std::shared_ptr<AvailabilityZoneManager>;
+
+    virtual ~AvailabilityZoneManager() = default;
+
+    virtual AvailabilityZone& get_zone(const std::string& name) = 0;
+    virtual std::vector<std::reference_wrapper<const AvailabilityZone>> get_zones() = 0;
```

Comment:
I wouldn't mind an include for the reference_wrapper, nor for vector, string, and the smart ptrs while you're at it.